lib/viewmodels/login_view_model.dart

```diff
@@ -21,13 +22,28 @@ final isUserLoggedInProvider = Provider<bool>((ref) {
   return ref.watch(userProvider).valueOrNull != null;
 });
 
-/// Firebase logged in user id provider
+/// Firebase logged in user id provider, returns null if the user is not logged
+/// in
```

Comment:
I think adding some small warning can be interesting:
```suggestion
/// in. [validUidProvider] should almost always be used: its
/// error does not cause a pop-up to be shown by the circular
/// value, which is typically what one want since the user will
/// get navigated back to the log-in page anyway.
```